src/detect-engine-address.c

```diff
@@ -1678,37 +1677,37 @@ int DetectAddressMatchIPv6(const DetectMatchAddressIPv6 *addrs,
         SCReturnInt(0);
     }
 
+    uint32_t match_addr[4];
+    match_addr[0] = SCNtohl(a->addr_data32[0]);
+    match_addr[1] = SCNtohl(a->addr_data32[1]);
+    match_addr[2] = SCNtohl(a->addr_data32[2]);
+    match_addr[3] = SCNtohl(a->addr_data32[3]);
+
     /* See if the packet address is within the range of any entry in the
      * signature's address match array.
      */
     for (uint16_t idx = 0; idx < addrs_cnt; idx++) {
         uint16_t result1 = 0, result2 = 0;
 
         /* See if packet address equals either limit. Return 1 if true. */
-        if (SCNtohl(a->addr_data32[0]) == addrs[idx].ip[0] &&
-            SCNtohl(a->addr_data32[1]) == addrs[idx].ip[1] &&
-            SCNtohl(a->addr_data32[2]) == addrs[idx].ip[2] &&
-            SCNtohl(a->addr_data32[3]) == addrs[idx].ip[3])
-        {
+        if (match_addr[0] == addrs[idx].ip[0] && match_addr[1] == addrs[idx].ip[1] &&
```

Comment:
wonder if we should just use `memcmp` here, at 16 bytes its ideal for simd optimizations (though we might not be aligned correctly, but that will be handled by libc)

Reply:
Good point ... this can be done.

Reply:
Do you know if it is worth a function call ?